src/taskscheduler.cpp

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "taskscheduler.h"
+#include "leakdetector.h"
+#include "logger.h"
+#include "mozillavpn.h"
+#include "task.h"
+
+namespace {
+Logger logger(LOG_MAIN, "TaskScheduler");
+TaskScheduler* s_taskScheduler = nullptr;
+}  // namespace
+
+// static
+void TaskScheduler::scheduleTask(Task* task) {
+  Q_ASSERT(task);
+  logger.debug() << "Scheduling task: " << task->name();
+  maybeCreate()->scheduleTaskInternal(task);
+}
+
+// static
+void TaskScheduler::deleteTasks() { maybeCreate()->deleteTasksInternal(); }
+
+// static
+TaskScheduler* TaskScheduler::maybeCreate() {
+  if (!s_taskScheduler) {
+    s_taskScheduler = new TaskScheduler(MozillaVPN::instance());
+  }
+  return s_taskScheduler;
+}
+
```

Comment:
You can use the new C++ magic static here.  It is also thread safe (for free).  Here is a little article on it:
http://www.nuonsoft.com/blog/2017/08/10/implementing-a-thread-safe-singleton-with-c11-using-magic-statics/comment-page-1/